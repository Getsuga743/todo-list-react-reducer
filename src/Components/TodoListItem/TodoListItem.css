.list-item-container {
  color: hsl(235, 19%, 35%);
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  text-align: center;
  align-items: center;
  background-color: #ffffff;
  border-bottom: 0.5px solid hsl(233, 11%, 84%);
  width: 100%;
  word-spacing: 0.1rem;
}
.list-item {
  color: hsl(235, 19%, 35%);
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  text-align: center;
  align-items: center;
  background-color: #ffffff;
}
.list-item-content {
  display: flex;
  width: 100%;
  justify-content:space-between;
}
.list-item-content p{
    width:100%;
    text-align: start;
}
.time{
    margin:0 2rem;
}
.list-item:last-child {
  border: none;
}
.item-index {
  display: flex;
  width: 1.4rem;
  height: 1.4rem;
  margin:0rem;  
  border-radius: 50%;
  overflow:hidden;
}
.item-index-span{
  width:100%;
  height:100%;
  display: flex;
  background: linear-gradient( rgb(87, 221, 255) , rgb(192, 88, 243) 70.71%);
  justify-content: center;
  align-items: center;
}
.item-index span svg{
  position:relative;
  left:0px;
}
.span-container{
  display: flex;
  justify-content:center;
  align-items: center;
  margin: 0 1.5rem;
}
.list-item-btn{
    display:flex;
    justify-content: center;
    align-items: center;
}
.border-hidden{
  border:1px solid hsl(236, 9%, 61%);
}
.btn.btn-item{
  width:4rem;
  height:100%;
  border: none;
  color: hsl(235, 19%, 35%);
  background-color: #ffffff;
  font-size: 1.5rem;
  z-index:5;
}

.hidden {
  display:hidden!important;
}