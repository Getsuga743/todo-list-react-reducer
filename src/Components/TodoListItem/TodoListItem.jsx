/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable react/prop-types */
import React from 'react';
import './TodoListItem.css';
import img from '../../images/icon-cross.svg';

const TodoListItem = ({ todo, handleToggle, handleDelete }) => (
  <>
    <li className="list-item-container">
      <span className="span-container">
        <span className={todo.done === true ? 'item-index' : ' item-index border-hidden'}>
          <span className={todo.done === true ? 'item-index-span' : ''}>
            <svg xmlns="http://www.w3.org/2000/svg" width="11" height="9">
              <path
                fill="none"
                stroke="#FFF"
                strokeWidth="2"
                d="M1 4.304L3.696 7l6-6"
              />
            </svg>
          </span>
        </span>
      </span>
      <div className="list-item-content">
        <p
          // se puede implementar con un template literal y un && ${todo.done && 'complete'}
          className={todo.done === false ? '' : 'done'}
          onClick={() => {
            handleToggle(todo.id);
          }}
        >
          {todo.description}
        </p>
        <div className="list-item-btn">
          <span className="time">{todo.time}</span>

          <button
            className="btn btn-item"
            onClick={() => {
              handleDelete(todo.id);
            }}
            type="button"
          >
            <img src={img} alt="delete item cross" />
          </button>
        </div>
      </div>
    </li>
  </>
);

export default TodoListItem;
