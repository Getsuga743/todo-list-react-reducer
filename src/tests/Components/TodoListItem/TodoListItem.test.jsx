/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import TodoListItem from '../../../Components/TodoListItem/TodoListItem';
import '../../../setupTests';
import demoTodos from '../../../fixtures/demoTodos';

describe('pruebas en <TodoListItem />', () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
    />,
  );
  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de llamar la funcion borrar', () => {
    wrapper.find('button').simulate('click');
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test('debe de llamar la funcion toggle', () => {
    wrapper.find('p').simulate('click');
    expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test('el contenido debe ser correcto', () => {
    wrapper.update();
    const p = wrapper.find('p');
    expect(p.text()).toBe(`${demoTodos[0].description}`);
  });
  test('debe de tener la clase complete si el  todo.done === true', () => {
    const todo = demoTodos[0];
    todo.done = true;
    const wrapper1 = shallow(<TodoListItem todo={todo} />);
    expect(wrapper1.find('p').hasClass('done')).toBe(true);
  });
});
