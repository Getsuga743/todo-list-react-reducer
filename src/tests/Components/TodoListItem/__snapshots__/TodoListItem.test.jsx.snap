// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <TodoListItem /> debe mostrarse correctamente 1`] = `
<Fragment>
  <li
    className="list-item-container"
  >
    <span
      className="span-container"
    >
      <span
        className=" item-index border-hidden"
      >
        <span
          className=""
        >
          <svg
            height="9"
            width="11"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 4.304L3.696 7l6-6"
              fill="none"
              stroke="#FFF"
              strokeWidth="2"
            />
          </svg>
        </span>
      </span>
    </span>
    <div
      className="list-item-content"
    >
      <p
        className=""
        onClick={[Function]}
      >
        Aprender React
      </p>
      <div
        className="list-item-btn"
      >
        <span
          className="time"
        />
        <button
          className="btn btn-item"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="delete item cross"
            src="icon-cross.svg"
          />
        </button>
      </div>
    </div>
  </li>
</Fragment>
`;
